src/adagrams.js

```diff
@@ -1,15 +1,116 @@
+// import { LETTER_POOL } from '../test/adagrams.test.js';
+
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterList = [];
+  for (const [letter, frequency] of Object.entries(LETTER_POOL)) {
+    for (let i = 0; i < frequency; i++) {
+      letterList.push(letter);
+    }
+  }
+
+  const lettersInHand = [];
+  
+  while (lettersInHand.length < 10) {
+    const index = Math.floor(Math.random() * letterList.length);
+    lettersInHand.push(letterList[index]);
+    letterList.splice(index, 1);
+  }
+  return lettersInHand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const lettersInHandMap = new Map();
+
+  // Populate the Map with letter counts
+  for (const letter of lettersInHand) {
+    if (lettersInHandMap.has(letter)) {
+      lettersInHandMap.set(letter, lettersInHandMap.get(letter) + 1);
+    } else {
+      lettersInHandMap.set(letter, 1);
+    }
+  }
+
+  // Check if the input can be formed with the letters in hand
+  for (const letter of input) {
+    if (!lettersInHandMap.has(letter) || lettersInHandMap.get(letter) === 0) {
+      return false;
+    } else {
+      lettersInHandMap.set(letter, lettersInHandMap.get(letter) - 1);
+    }
+  }
+
+  return true;
+};
+
+const scoreChart = {
+  A: 1, E: 1, I: 1, O: 1, U: 1, L: 1, N: 1, R: 1, S: 1, T: 1,
+  D: 2, G: 2,
+  B: 3, C: 3, M: 3, P: 3,
+  F: 4, H: 4, V: 4, W: 4, Y: 4,
+  K: 5,
+  J: 8, X: 8,
+  Q: 10, Z: 10,
 };
```

Comment:
If the values in the object are spread across multiple lines, list the first value indented along with the remaining values. And since JS doesn't care what order the keys are listed, it would be easier for another coder reading this if the letters were written alphabetically, as it makes it easier to confirm that all the letters are present.

Since this is a global constant, prefer using all caps for the name.
```js
const SCORE_CHART = {
  A: 1,
  B: 3,
  C: 3,
  ...
};
```